### PR TITLE
Feature/141 solr search

### PR DESCRIPTION
**Pull request by njcameron**
- solr modules
- solr config
- new view to replace search with sol
- new panel page to use solr search on page
- update to templates to allow extra content types in search listings
- add "line text" field to FAQ / news

### CODE REVIEW

File: www/modules/contrib/devel/log.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{"level":"debug","msg":"main.go started","time":"2017-03-24T19:33:53+01:00"}
```
Comment (annikaC) [inline www/modules/contrib/devel/log.json]: probably shouldn't commit this file? (apologies if you already got rid of it, just going through commits)

File: www/modules/custom/upd_search/upd_search.module [hunk 2]
```diff
@@ -42,18 +42,18 @@ function upd_search_preprocess_views_view(&$variables) {
...
+  foreach ($view->result as $result) {
+    $excerpt[$result->search_api_id] = !empty($result->search_api_excerpt) ? $result->search_api_excerpt : '';
+  }
   foreach ($variables['rows'] as $row) {
     foreach ($row['#rows'] as $item) {
       $id = 'entity:node/' . $item['#node']->id() . ':en';
-//      $item['#node']->excerpt = !empty($excerpt[$id]) ? $excerpt[$id] : upd_search_node_generate_excerpt($item['#node']);
-      $item['#node']->excerpt = upd_search_node_generate_excerpt($item['#node']);
+      $item['#node']->excerpt = !empty($excerpt[$id]) ? $excerpt[$id] : upd_search_node_generate_excerpt($item['#node']);
+//      $item['#node']->excerpt = upd_search_node_generate_excerpt($item['#node']);
```
Comment (annikaC) [inline www/modules/custom/upd_search/upd_search.module]: Let's not leave any commented code in if we can help it, makes it hard to read

File: www/modules/custom/upd_search/upd_search.module [hunk 3]
```diff
@@ -51,8 +56,8 @@ function upd_search_preprocess_views_view(&$variables) {
 //  }
   foreach ($variables['rows'] as $row) {
     foreach ($row['#rows'] as $item) {
-      $id = 'entity:node/' . $item['#node']->id() . ':en';
-//      $item['#node']->excerpt = !empty($excerpt[$id]) ? $excerpt[$id] : upd_search_node_generate_excerpt($item['#node']);
+      //$id = 'entity:node/' . $item['#node']->id() . ':en';
+      //$item['#node']->excerpt = !empty($excerpt[$id]) ? $excerpt[$id] : upd_search_node_generate_excerpt($item['#node']);
```
Comment (alfraser86) [inline www/modules/custom/upd_search/upd_search.module]: yeah lets get rid of old comments

File: www/modules/contrib/devel/log.json (hunk 1 above)
Comment (njcameron) [inline www/modules/contrib/devel/log.json]: Good catch, thanks

File: www/modules/custom/upd_search/upd_search.module (hunk 3 above)
Comment (njcameron) [inline www/modules/custom/upd_search/upd_search.module]: 👍

**Pull request merged**